packages/block-editor/src/hooks/custom-fields.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { PanelBody, TextControl } from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+import { hasBlockSupport } from '@wordpress/blocks';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { InspectorControls } from '../components';
+import { useBlockEditingMode } from '../components/block-editing-mode';
+
+/**
+ * Filters registered block settings, extending attributes to include `connections`.
+ *
+ * @param {Object} settings Original block settings.
+ *
+ * @return {Object} Filtered block settings.
+ */
+export function addAttribute( settings ) {
+	if ( hasBlockSupport( settings, 'connections', true ) ) {
+		// Gracefully handle if settings.attributes is undefined.
+		settings.attributes = {
+			...settings.attributes,
+			connections: {
+				type: 'object',
+			},
+		};
+	}
+
+	return settings;
+}
+
+/**
+ * Override the default edit UI to include a new block inspector control for
+ * assigning a connection to blocks that has support for connections.
+ * Currently, only the `core/paragraph` block is supported and there is only a relation
+ * between paragraph content and a custom field.
+ *
+ * @param {WPComponent} BlockEdit Original component.
+ *
+ * @return {WPComponent} Wrapped component.
+ */
+export const withInspectorControl = createHigherOrderComponent(
+	( BlockEdit ) => {
+		return ( props ) => {
+			const blockEditingMode = useBlockEditingMode();
+			const hasCustomFieldsSupport = hasBlockSupport(
+				props.name,
+				'connections',
+				false
+			);
+			// We prevent that the content is lost when the user removes the custom field.
+			// Editing the content in the paragraph block with a placeholder is not the best solution.
```

Comment:
I should definitely find an alternative rather than messing with the paragraph content.

Reply:
Not sure what you had in mind, but having a placeholder (like you have) but non-editable should be enough for the first version. If you connect the content attribute to a custom field, it will actually replace the inner content of the paragraph.

In a future phase, when users can edit the value of the custom field directly through the editor, we'll have to figure out the best UI and UX. But so far, a placeholder that replaces the paragraph's content seems fine to me. Keep in mind that this is just an experimental MVP.

Reply:
> Not sure what you had in mind, but having a placeholder (like you have) but non-editable should be enough for the first version.

Let's do that and check that if the user changes its content and the meta_key field for several times in a row we don't mess with the paragraph content.

Reply:
After spending some time with this. For a MVP is OK to leave it as is. It prevents losing the first paragraph content. And, if you edit it again, then in that case that new edition will be lost if you edit the field again.